packages/hash/frontend/src/blocks/page/createEditorView.ts

```diff
@@ -123,9 +123,13 @@ export const createEditorView = (
 
   view.dom.classList.add(styles.ProseMirror);
 
-  for (const meta of preloadedBlocks) {
-    manager.defineNewBlock(meta);
-  }
+  // prosemirror will use the first node type (per group) for auto-creation.
+  // we want this to be the paragraph node type.
+  Object.values(blocksMeta)
+    .sort((blockMeta) =>
+      blockMeta.componentMetadata.name === "paragraph" ? -1 : 0,
+    )
+    .forEach(manager.defineNewBlock, manager);
```

Comment:
*Good*: I like the comment, thanks for adding it. Makes it a lot less implicit / more clear. This change also means the paragraph-first behavior is enforced closer to where its needed, rather than where the object is defined, and we don't depend on key order. Great! 

*Room for improvement*: I think the functional style of this does make it slightly less readable, and it is not necessarily optimised for change. 

We don't need to sort the metas to ensure paragraph is defined first – because it checks if the node has been defined and does nothing if it has been. We can just manually define paragraph, and then loop through the rest.

That might look something like this

```suggestion
  const paragraphUrl = "https://block.blockprotocol.org/paragraph";
  if (blocksMeta[paragraphUrl]) {
    manager.defineNewBlock(blocksMeta[paragraphUrl]);
  }
  
  for (const blockMeta of Object.values(blocksMeta)) {
    manager.defineNewBlock(blockMeta);
  }
```

We can change `paragraphUrl` once the URLs in `blockPaths.json` are a bit nicer.

Reply:
I am not too comfortable hard-coding a url here so I'll go with the `componentMetadata.name` instead.

Reply:
Done.